### ISSUE TITLE
Test Character List against GW Data etc

### ISSUE THREAD

**Issue by mac-nw** [Phr33d0m/NW-Profession-Bot#47]
After writing about the client.dataModel.model.loginInfo.choices character object list in the other issue, i realize we should be using this to test characters added to the bot and for the bank char field.
#### ON settings save:
1. Test all char names to make sure they exit in the gw data model.
2. If data does not match, cancel the save and prompt to fix the problematic character names.
#### BEFORE character switch:
1. Check if next character is in the gw data model list.
2. If Character is missing remove it from professions and reload professions. (Perhaps save a message to be prompted after reload).

Suggestions welcome ^_^

**Comment by noonereally** [comment]
If the data model has all the character names, maybe with the settings change you suggested (to a more structured JSON and not a simple array) it is possible to get rid of the character input all togheder.

Just need to add a flag to settings if the char is active / not active (if the script should do a pass on it or ignore)

then:
Script load, login.
Compare names in model vs saved in setting.
    1. delete those names that are in settings and not in the model.  (deleted/renamed old characters)
    2. add from model those that missing in settings (new or name changed) and mark them non active
    so the user can set their configurations in the UI

the active non active is a checkbox in the char  settings part and all users names should be shown in the UI

Can be done even with the current settings array and can fix the char deletion problem (with the missing number in the loops).
And save space and errors with the number of char input and names input.
Perhaps a notification when loading and all users are marked as non active that the user need to go and set their tasks and settings / enable those he/she wants the script to work with (will happen on first load)

**Comment by RottenMind** [comment]
Nothing to add here, because its still unclear, how that one user got "settings wipe" happen after he´s  VMware machine rebooted.(VMware must cache something what did make wipe happen)

..but good anyway that he got that happen before it created bigger issue.

**Comment by noonereally** [comment]
Tried it on something small (the refine counters) and looks to me that 
client.dataModel.model.loginInfo.choices
can replace the whole char input, but it will take work to change it all

I've put the check into the login function (in the fork)

**Comment by RottenMind** [comment]
Look link, its not advanced as this bot, but it has two unique options,
- wipe one character by name from settings(not sure if it usefull to this)
- open console log on browser window(off topic)

https://greasyfork.org/en/forum/discussion/3115/x

**Comment by Phr33d0m** [comment]
@mac-nw I like what @noonereally suggested about removing the input stuff and just populate with all the available characters from that model and then just a flag to enable/disable a character.

A small issue with that though. That seems to not update that regularly (it caches on the server side). So if you've just created 10 characters you might need to wait a couple of hours for them to appear in that model... But well, I can live with that ;-)

**Comment by noonereally** [comment]
I tried something in that spirit
https://github.com/noonereally/NW-Profession-Bot/commit/4125030ece4ffba81ec92e69b817a6bc9aa86fb4
but the frequent writes crushed chrome, need to add flag that checks if settings already loaded for that 'session' and skip the loads, i'm not sure why this function runs each second or so and not only on load. And the frequent writen stuff should be separate like the counters now. 
(I've reverted that change later, it's a lot of work, I don't have the time to do the whole thing)

Also, this way, the settings are saved per account, which is an added bonus in my opinion.
